makefiles/tools/openocd-adapters/stlink.inc.mk

```diff
@@ -5,7 +5,7 @@ STLINK_VERSION ?= 2-1
 
 # Determine which configuration file to use for OpenOCD.
 # Also pass the correct version of the ST-Link adapter to the script. 
-OPENOCD_ADAPTER_INIT ?= -c 'source $(RIOTBASE)/dist/tools/openocd/adapters/stlink.cfg ; set_stlink_cfg $(STLINK_VERSION)' 
+OPENOCD_ADAPTER_INIT ?= -c 'set stlink_version $(STLINK_VERSION) ; source $(RIOTBASE)/dist/tools/openocd/adapters/stlink.cfg' 
```

Comment:
This method sets a variable that makes sense and sources the file at the same level as the original used. 
Adding a procedure to select the correct config adds extra complexity and risk.